src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -216,7 +217,7 @@ type StatePatchBody = {
 		is_on__commit?: string | null;
 		apps?: Array<{
 			services?: {
-				[serviceId: string]: {
+				[imageId: string]: {
```

Comment:
> Is this correct? Does the Supervisor really send image IDs as keys in a map called services?

Yes, that is correct 

https://github.com/balena-os/balena-supervisor/blob/master/src/compose/application-manager.ts#L844-L850

Reply:
I'm not sure about the breaking part, but this looks like just a rename on the type?

Reply:
> With "breaking" I meant that the expectation (based on the type) seemed to be this was a service ID but apparently with this change we're now treating it as image ID

I see, since the type was not exported before and it looks like the only place that key was used (as imageId) was [here](https://github.com/balena-io/open-balena-api/blob/8de3a73e17492d10120d0b20d277d74020328d1e/src/features/device-state/routes/state-patch.ts#L427) this might not be breaking at all

> However -- just curious -- how did we end up sending image IDs from the Supervisor in map called services

Not sure about the logic of reporting the service like that. That mapping seems to be there since the module was first written in the supervisor https://github.com/balena-os/balena-supervisor/commit/cb37f7ebcc20125c6aa13f954ba38c8b91c40eda#diff-47198ccd5f456a10e7e3b9f75194ac1d47fa5e450a122cd2e56df3620bd705f8R222

Reply:
Right, just the type was wrong all along then. Thanks for looking all this up!

Reply:
Yep, I changed it because when I was looking at the supervisor I realized that it was indeed using the imageId and I updated the informational type name to match what it actually represents (it's not the first case I've had to do similar updates to match reality since the code predates the usage of typescript and so the types have to match to the unintended reality that emerged)